### PR TITLE
issue #2 and issue 3

### PR DESCRIPTION
**Pull request by xindiguo**

### CODE REVIEW

File: pythonSynapseUtils/auditAndUpdateByCommonDict.py [hunk 1]
```diff
@@ -63,25 +62,25 @@ def _helperAuditCommonDict(syn, temp, annoDict,entityMissAllAnno,incorrectAnnota
             for x in removed:
                 annoKey = str(x)
                 if annoKey in missingAnno.keys():
-                    missingAnno[annoKey].append(tempId)
+                    missingAnno[annoKey].append(temp)
```
Comment (kdaily) [inline pythonSynapseUtils/auditAndUpdateByCommonDict.py]: more descriptive name than `temp` - maybe `currSynObj`?

Comment (kdaily) [inline pythonSynapseUtils/auditAndUpdateByCommonDict.py]: use `defaultdict`

File: pythonSynapseUtils/auditAndUpdateByCommonDict.py [hunk 2]
```diff
@@ -40,20 +39,20 @@ def auditCommonDict(syn, synId, annoDict):
...
                 _helperAuditCommonDict(syn,temp,annoDict,entityMissAllAnno,incorrectAnnotated,missingAnno)
-        return entityMissAllAnno,incorrectAnnotated,missingAnno
+        yield entityMissAllAnno,incorrectAnnotated,missingAnno
                 
 def _helperAuditCommonDict(syn, temp, annoDict,entityMissAllAnno,incorrectAnnotated,missingAnno):
     print "Checking..."
     tempDict = temp.annotations
-    tempId = temp.id
+
     if bool(tempDict):
```
Comment (kdaily) [inline pythonSynapseUtils/auditAndUpdateByCommonDict.py]: don't need `bool`

File: pythonSynapseUtils/auditAndUpdateByCommonDict.py (hunk 1 above)
Comment (kkdang) [inline pythonSynapseUtils/auditAndUpdateByCommonDict.py]: @kdaily says line 62 check for len(removed) is redundant.

File: pythonSynapseUtils/auditAndUpdateByCommonDict.py [hunk 3]
```diff
@@ -143,7 +142,7 @@ def updateAnnoByIdDictFromDict(syn,idDict,annoDict):
     
     """
     Update annotations from dictionary
-    by a given dict(key:annotation keys need to updated, value:a list of Synpase IDs)
+    by a given dict(key:annotation keys need to updated, value:a list of Synpase Objects)
```
Comment (kkdang) [inline pythonSynapseUtils/auditAndUpdateByCommonDict.py]: For readability, consider adding another function argument -- a flag for whether input is list of syn entity objects or not. Could set default = True.

File: pythonSynapseUtils/auditAndUpdateByCommonDict.py [hunk 4]
```diff
@@ -158,7 +157,7 @@ def updateAnnoByIdDictFromDict(syn,idDict,annoDict):
         print "updating annotaion values for key: %s" % key
         for synId in idDict[key]:
             print "> %s" %synId
-            temp = syn.get(synId, downloadFile = False)
+            temp = synId
```
Comment (kkdang) [inline pythonSynapseUtils/auditAndUpdateByCommonDict.py]: Don't need to add new variable here. Consider changing synID name to synEntity.

**Pull request closed without merging**